src/com/mathgame/cards/NumberCard.java

```diff
@@ -98,7 +104,75 @@ public String getHome() {
 	public void setHome(String home) {
 		this.home = home;
 	}
-
+	/**
+	 * Converts a string with a zero after the decimal point
+	 * and truncates it, along with the decimal point 
+	 * @param s - the string to evaluate
+	 * @return the same string or one with a truncated zero
+	 */
+	public static String truncZero(String s){
+		
+		if(s.contains(".")){
+			int dpPos = 0;
+			String result = "";
+			for(int i = 0; i < s.length(); i++){
+				
+				if(s.charAt(i)=='.'){
+					dpPos = i;
+					break;
+				}
+				result += s.charAt(i);
+			}
+			if(dpPos==s.length()-1)
+				return result;
+			for(int i = dpPos+1; i < s.length(); i++)
+				if(s.charAt(i)!='0')
+					return s;
+			System.out.println("Zero truncated");
+			return result;
+		}
+		else return s;
+	}
+	public static String parseLatex(String s){
```

Comment:
Better yet, update the database files to format the strings in LateX format so you don't need to do all this parsing!